packaging/tests/tas-restapi/src/test/java/org/alfresco/rest/rules/RuleSetLinksTests.java

```diff
@@ -0,0 +1,313 @@
+    public void linkToNonExistingRuleSet()
+    {
+        STEP("Create a folder in existing site");
+        final FolderModel folder = dataContent.usingUser(user).usingSite(site).createFolder();
+
+        STEP("Link to non-existing rule set");
+        final RestRuleSetLinkModel request = new RestRuleSetLinkModel();
+        request.setId("dummy-rule-set-id");
+        restClient.authenticateUser(user).withCoreAPI().usingNode(folder).createRuleLink(request);
+
+        STEP("Assert link result is 404");
+        restClient.assertStatusCodeIs(NOT_FOUND);
+    }
+
+    /**
+     * Check we get bad request error when linking to a folder without rules.
+     */
+    @Test(groups = {TestGroup.REST_API, TestGroup.RULES})
+    public void linkToFolderWithoutRules()
+    {
+        STEP("Create 2 folders without rules in existing site");
+        final FolderModel folder1 = dataContent.usingUser(user).usingSite(site).createFolder();
+        final FolderModel folder2 = dataContent.usingUser(user).usingSite(site).createFolder();
+
+        STEP("Link to a folder without rules");
+        final RestRuleSetLinkModel request = new RestRuleSetLinkModel();
+        request.setId(folder2.getNodeRef());
+        restClient.authenticateUser(user).withCoreAPI().usingNode(folder1).createRuleLink(request);
+
+        STEP("Assert link result is 400");
+        restClient.assertStatusCodeIs(BAD_REQUEST)
+                .assertLastError().containsSummary("The target node has no rules to link.");
+    }
+
+    /**
+     * Check we get bad request error when linking from a folder which already has rules.
+     */
+    @Test(groups = {TestGroup.REST_API, TestGroup.RULES})
+    public void linkFromFolderWithRules()
+    {
+        STEP("Create folders in existing site");
+        final FolderModel folder1 = dataContent.usingUser(user).usingSite(site).createFolder();
+        final FolderModel folder2 = dataContent.usingUser(user).usingSite(site).createFolder();
+
+        STEP("Create rules in both folders.");
+        RestRuleModel ruleModel1 = createRuleModel("ruleName1");
+        restClient.authenticateUser(user).withCoreAPI().usingNode(folder1).usingDefaultRuleSet()
+                .createSingleRule(ruleModel1);
+        RestRuleModel ruleModel2 = createRuleModel("ruleName2");
+        restClient.authenticateUser(user).withCoreAPI().usingNode(folder2).usingDefaultRuleSet()
+                .createSingleRule(ruleModel2);
+
+        STEP("Link from a folder with rules");
+        final RestRuleSetLinkModel request = new RestRuleSetLinkModel();
+        request.setId(folder2.getNodeRef());
+        restClient.authenticateUser(user).withCoreAPI().usingNode(folder1).createRuleLink(request);
+
+        STEP("Assert link result is 400");
+        restClient.assertStatusCodeIs(BAD_REQUEST)
+                .assertLastError().containsSummary(
+                "Unable to link to a ruleset because the folder has pre-existing rules or is already linked to a ruleset.");
```

Comment:
We should fix this to say "rule set", since we've standardised on that everywhere else.

Reply:
Since this is a comm-repo codebase change and we have code freeze now, I'll create a ticket to do that later on.